libraries/nodejs_install.rb

```diff
@@ -0,0 +1,61 @@
+#
+# Cookbook Name:: languages
+# HWRP:: nodejs_install
+#
+# Copyright 2014, Chef Software, Inc.
+#
+# Licensed under the Apache License, node_version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require 'open-uri'
+
+class Chef
+  class Resource::NodejsInstall < ChefCompat::Resource
```

Comment:
I left it this was since it was easy enough to change so the rest of the team could weigh in on pros/cons. You said yesterday that you preferred the HWRP way but we could discuss it during the working agreement so I continued the way I had it written.